metals/src/main/scala/scala/meta/internal/metals/ProcessOutput.scala

```diff
@@ -0,0 +1,74 @@
+package scala.meta.internal.metals
+
+import java.lang.StringBuilder
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+import fansi.ErrorMode
+
+/**
+ * Handles streaming console output from a system process with potential ANSI codes.
+ *
+ * @param onLine The callback handler when the process has printed a single line.
+ *               Guaranteed to have no newline \n characters.
+ */
+class ProcessOutput(onLine: String => Unit) {
```

Comment:
De facto a `LineListener`. Let's keep this abstraction simple - it does not know anything about the source of those lines, nor should it. Therefore we can drop the `Process` from the name.
Also, this name would play nicely with the name `onLine` callback.